From ok@artecdesign.ee Fri Nov 25 16:01:02 2005
Date: Sat, 26 Nov 2005 01:58:36 +0200 (EET)
From: Olav Kongas <ok@artecdesign.ee>
To: Greg KH <greg@kroah.com>
Subject: isp116x-hcd: minor cleanup
Message-ID: <Pine.LNX.4.63.0511260153120.21074@pcy.artec.ee>

When going to suspend, there's no point in setting HC state in
host controller driver as USB core takes care of this.

Signed-off-by: Olav Kongas <ok@artecdesign.ee>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/isp116x-hcd.c |    2 --
 1 file changed, 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/host/isp116x-hcd.c
+++ gregkh-2.6/drivers/usb/host/isp116x-hcd.c
@@ -1426,7 +1426,6 @@ static int isp116x_bus_suspend(struct us
 	val = isp116x_read_reg32(isp116x, HCCONTROL);
 	switch (val & HCCONTROL_HCFS) {
 	case HCCONTROL_USB_OPER:
-		hcd->state = HC_STATE_QUIESCING;
 		val &= (~HCCONTROL_HCFS & ~HCCONTROL_RWE);
 		val |= HCCONTROL_USB_SUSPEND;
 		if (device_may_wakeup(&hcd->self.root_hub->dev))
@@ -1434,7 +1433,6 @@ static int isp116x_bus_suspend(struct us
 		/* Wait for usb transfers to finish */
 		mdelay(2);
 		isp116x_write_reg32(isp116x, HCCONTROL, val);
-		hcd->state = HC_STATE_SUSPENDED;
 		/* Wait for devices to suspend */
 		mdelay(5);
 	case HCCONTROL_USB_SUSPEND:
